portalnet/src/discovery.rs

```diff
@@ -264,6 +264,14 @@ impl Discovery {
         self.node_addr_cache.write().get(node_id).cloned()
     }
 
+    /// Put a `NodeAddress` into cache. If the key already exists in the cache, then it updates the key's value and
+    /// returns the old value. Otherwise, `None` is returned.
+    pub fn put_cached_node_addr(&self, node_addr: NodeAddress) -> Option<NodeAddress> {
```

Comment:
Looks like we're never calling this, besides in your test? Is there somewhere else in the production code that will update the `node_addr_cache`?

Reply:
Not sure about a use case to add a node to the cache outside of discovery, besides in testing. The other way around is to expose the cache and make it public but I think it is more foolproof to add a discovery method for it.